src/App.tsx

```diff
@@ -1,27 +1,76 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
-
-// import { getAll, get5First, getRed } from './api/goods';
-// or
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
+// or import { getAll, get5First, getRed } from './api/goods';
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [allUsers, setAllUsers] = useState<Good[]>([]);
+  const [allClicked, setAllClicked] = useState(false);
+  const [fiveUsers, setFiveUsers] = useState<Good[]>([]);
+  const [load5Clicked, setLoad5Clicked] = useState(false);
+  const [redUsers, setRedUsers] = useState<Good[]>([]);
+  const [redClicked, setRedClicked] = useState(false);
+
+  useEffect(() => {
+    getAll().then(setAllUsers);
+    get5First().then(setFiveUsers);
+    getRedGoods().then(setRedUsers);
+  }, [allClicked, load5Clicked, redClicked]);
+
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
+
+      <button
+        type="button"
+        data-cy="all-button"
+        onClick={() => {
+          setRedClicked(false);
+          setLoad5Clicked(false);
+          setAllClicked(true);
+        }}
+      >
+        Load all goods
+      </button>
+
+      <button
+        type="button"
+        data-cy="first-five-button"
+        onClick={() => {
+          setAllClicked(false);
+          setRedClicked(false);
+          setLoad5Clicked(true);
+        }}
+      >
+        Load 5 first goods
+      </button>
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+      <button
+        type="button"
+        data-cy="red-button"
+        onClick={() => {
+          setAllClicked(false);
+          setLoad5Clicked(false);
+          setRedClicked(true);
+        }}
+      >
+        Load red goods
+      </button>
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      {allClicked && (
+        <GoodsList goods={allUsers} />
+      )}
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      {load5Clicked && (
+        <GoodsList goods={fiveUsers} />
+      )}
 
-    <GoodsList goods={[]} />
-  </div>
-);
+      {redClicked && (
+        <GoodsList goods={redUsers} />
+      )}
```

Comment:
it does the job, yet it's over-engineered @mmahfuzi 🙂 

it makes more sense to render `GoodsList` only once and pass the relevant array inside
```
const [fetchedData, setFetchedData] = useState...;

const fetch5 = async () => {
  apiMethodFetch5().then(setFetchedData)
}
const fetchAll = async () => {
  apiMethodFetchAll().then(setFetchedData)
}
const fetchRed = async () => {
  apiMethodFetchRed().then(setFetchedData)
}

return (
  <button onClick={fetch5} ...
  <button onClick={fetchAll} ...
  <button onClick={fetchRed} ...

  ...
  <GoodsList goods={fetchedData} />
```